odoo/addons/base/wizard/base_language_install.py

```diff
@@ -9,45 +9,56 @@ class BaseLanguageInstall(models.TransientModel):
 
     @api.model
-    def _default_language(self):
+    def _default_lang_ids(self):
         """ Display the selected language when using the 'Update Terms' action
             from the language list view
         """
         if self._context.get('active_model') == 'res.lang':
-            lang = self.env['res.lang'].browse(self._context.get('active_id'))
-            return lang.code
+            return self.env['res.lang'].browse(self._context.get('active_ids') or [self._context.get('active_id')])
         return False
-
-    @api.model
-    def _get_languages(self):
-        return [[code, name] for code, _, name, *_ in self.env['res.lang'].get_available()]
-
-    lang = fields.Selection(_get_languages, string='Language', required=True,
-                            default=_default_language)
+    # add a context on the field itself, to be sure even inactive langs are displayed
+    lang_ids = fields.Many2many('res.lang', 'res_lang_install_rel', 'language_wizard_id', 'lang_id', 'Languages', default=_default_lang_ids, context={'active_test': False})
     overwrite = fields.Boolean('Overwrite Existing Terms',
                                default=True,
                                help="If you check this box, your customized translations will be overwritten and replaced by the official ones.")
-    state = fields.Selection([('init', 'init'), ('done', 'done')],
-                             string='Status', readonly=True, default='init')
+    first_lang_id = fields.Many2one('res.lang', compute='_compute_first_lang_id', help="Used when the user only selects one language and is given the option to switch to it")
+
+    def _compute_first_lang_id(self):
+        for record in self:
+            lang_ids = record.lang_ids
+            record.first_lang_id = lang_ids[0] if lang_ids else False
 
     def lang_install(self):
         self.ensure_one()
         mods = self.env['ir.module.module'].search([('state', '=', 'installed')])
-        self.env['res.lang']._activate_lang(self.lang)
-        mods._update_translations(self.lang, self.overwrite)
-        self.state = 'done'
+        lang_ids = self.with_context(active_test=False).lang_ids
+        langs_to_activate = lang_ids.filtered(lambda l: not l.active)
+        langs_to_activate.toggle_active()
+        mods._update_translations(lang_ids.mapped('code'), self.overwrite)
         self.env.cr.execute('ANALYZE ir_translation')
 
+        if len(lang_ids) == 1:
+            return {
+                'type': 'ir.actions.act_window',
+                'res_model': 'base.language.install',
+                'res_id': self.id,
+                'view_mode': 'form',
+                'target': 'new',
+                'views': [[self.env.ref('base.language_install_view_form_lang_switch').id, 'form']],
+            }
+
+        message = _("The languages that you selected have been successfully installed. Users can choose their favorite language in their preferences.")
```

Comment:
OOC, why defining it here ? Could be inlined in the returned dict (and multi-lines with some python str multi line magic).